### ISSUE TITLE
Full EWMH 1.3 support

### ISSUE THREAD

**Issue by emersion** [swaywm/wlroots#1496]
Spec: https://specifications.freedesktop.org/wm-spec/wm-spec-1.3.html

Properties that MUST be supported:

- [ ] `_NET_SUPPORTED` (there are missing atoms here)
- [ ] `_NET_DESKTOP_VIEWPORT`
- [ ] `_NET_CURRENT_DESKTOP`
- [ ] `_NET_REQUEST_FRAME_EXTENTS`
- [x] `_NET_WM_STATE`
- [ ] `_NET_FRAME_EXTENTS`

**Comment by emersion** [comment]
No, this isn't possible, Wayland windows don't have X11 IDs.

Instead, clients should use the window-management protocol.

**Comment by chrisjbillington** [comment]
Ok no worries. I suppose we can hope for similar panel applications using the foreign-toplevel protocol at some point in the future then.

**Comment by progandy** [comment]
> Ok no worries. I suppose we can hope for similar panel applications using the foreign-toplevel protocol at some point in the future then.

https://github.com/WayfireWM/wf-shell has a dock with foreign-toplevel protocol.

**Comment by ammen99** [comment]
@progandy Sadly the dock now uses wayfire-shell because of issues with GTK tooltips and layer-shell. If somebody knows a workaround, I'd be happy to switch to layer-shell, at least for the dock.

**Comment by ddevault** [comment]
Hm, layer-shell supports xdg-popup. I'm surprised that it doesn't work.

**Comment by ddevault** [comment]
Unless, god forbid, GTK tooltips are xdg toplevels?

**Comment by ammen99** [comment]
> Unless, god forbid, GTK tooltips are xdg toplevels?

Yes, they are, if you use custom GTK backend for GtkWindow. It is a GTK bug though AFAIK.
